src/vm/vm_core.rs

```diff
@@ -666,8 +666,8 @@ impl VirtualMachine {
     }
 
     ///Makes sure that all assigned memory cells are consistent with their auto deduction rules.
-    pub fn verify_auto_deductions(&mut self) -> Result<(), VirtualMachineError> {
-        for (name, builtin) in self.builtin_runners.iter_mut() {
+    pub fn verify_auto_deductions(&self) -> Result<(), VirtualMachineError> {
```

Comment:
I'm confused here. Isn't the `deduce_memory_cell` supposed to mutate `self.memory`?

Reply:
The values obtained from `deduce_memory_cell` are inserted into memory in `compute_operands`, not here

Reply:
That makes sense, thanks!